src/EnergyPlus/WeatherManager.cc

```diff
@@ -3458,13 +3459,17 @@ namespace WeatherManager {
                     TomorrowLiquidPrecip(CurTimeStep, Hour) = LiquidPrecip;
                     TomorrowHorizIRSky(CurTimeStep, Hour) = IRHoriz;
 
-                    if (Environment(Envrn).WP_Type1 == 0) {
+                    bool SkyTempScheduled = false;
+
+                    if (Environment(Envrn).WP_Type1 != 0 && WPSkyTemperature(Environment(Envrn).WP_Type1).IsSchedule) {
+                        SkyTempScheduled = true;
+                    }
+
+                    if (!SkyTempScheduled) {
```

Comment:
Why all the extra gymnastics above? Seems that lines 3462:3466 could be eliminated and this line could be 
`if (!Environment(Envrn).WP_Type1==WP_ScheduleValue)`
Oh, but I see `WP_Type1` is the index of the WeatherProperty. So, seems like it would be more efficient to change `WP_Type1` to be `WP_SkyTempType` and set it when processing the WeatherProperty object? Then all of these ifs could be more direct.

Reply:
I added a `SkyTempModel` field in the `Environment`. It's still required in `WeatherProperty` object for matching with Environments and reading in the schedules.